server/service/src/login.rs

```diff
@@ -282,6 +282,14 @@ impl LoginService {
     }
 }
 
+fn optional_string(str: &str) -> Option<String> {
```

Comment:
This (TODO) is pretty simple (did a quick test seems to work)

```diff
diff --git a/server/service/src/apis/login_v4.rs b/server/service/src/apis/login_v4.rs
index 803b6a9d6..23cf87a97 100644
--- a/server/service/src/apis/login_v4.rs
+++ b/server/service/src/apis/login_v4.rs
@@ -1,3 +1,4 @@
+use crate::sync::sync_serde::empty_str_as_option_string;
 use reqwest::{Client, Url};
 use serde::{Deserialize, Serialize};
 
@@ -49,17 +50,22 @@ pub struct LoginUserV4 {
     // permissions_spare: null,
     pub lasttime: i32,
     pub initials: String,
-    pub first_name: String,
-    pub last_name: String,
+    #[serde(deserialize_with = "empty_str_as_option_string")]
+    pub first_name: Option<String>,
+    #[serde(deserialize_with = "empty_str_as_option_string")]
+    pub last_name: Option<String>,
     //date_of_birth: "0000-00-00",
     pub address_1: String,
     pub address_2: String,
-    pub e_mail: String,
-    pub phone1: String,
+    #[serde(deserialize_with = "empty_str_as_option_string")]
+    pub e_mail: Option<String>,
+    #[serde(deserialize_with = "empty_str_as_option_string")]
+    pub phone1: Option<String>,
     pub phone2: String,
     //date_created: "2017-10-11",
     //date_left: "0000-00-00",
-    pub job_title: String,
+    #[serde(deserialize_with = "empty_str_as_option_string")]
+    pub job_title: Option<String>,
     pub responsible_officer: bool,
     #[serde(rename = "Language")]
     pub language: i32,
diff --git a/server/service/src/login.rs b/server/service/src/login.rs
index 0f77c30e5..b8f06dcef 100644
--- a/server/service/src/login.rs
+++ b/server/service/src/login.rs
@@ -224,7 +224,7 @@ impl LoginService {
             username: user_info.user.name.to_string(),
             hashed_password: UserAccountService::hash_password(&password)
                 .map_err(UpdateUserError::PasswordHashError)?,
-            email: optional_string(&user_info.user.e_mail),
+            email: user_info.user.e_mail,
             language: match user_info.user.language {
                 0 => Language::English,
                 1 => Language::French,
@@ -236,10 +236,10 @@ impl LoginService {
                 7 => Language::Tetum,
                 _ => Language::English,
             },
-            first_name: optional_string(&user_info.user.first_name),
-            last_name: optional_string(&user_info.user.last_name),
-            phone_number: optional_string(&user_info.user.phone1),
-            job_title: optional_string(&user_info.user.job_title),
+            first_name: user_info.user.first_name,
+            last_name: user_info.user.last_name,
+            phone_number: user_info.user.phone1,
+            job_title: user_info.user.job_title,
         };
         let stores_permissions: Vec<StorePermissions> = user_info
             .user_stores
diff --git a/server/service/src/sync/mod.rs b/server/service/src/sync/mod.rs
index 8f59b413e..e8072fafa 100644
--- a/server/service/src/sync/mod.rs
+++ b/server/service/src/sync/mod.rs
@@ -8,7 +8,7 @@ pub mod settings;
 pub mod site_info;
 mod sync_api_credentials;
 mod sync_buffer;
-mod sync_serde;
+pub(crate) mod sync_serde;
 pub mod sync_status;
 pub mod synchroniser;
 pub mod synchroniser_driver;
```

Btw can apply above by copying in a file and `git apply 'filepath'`

Reply:
sweet! thanks